### ISSUE TITLE
Multipliciation of a `KroneckerProduct` with a `KroneckerProduct` of `Diagonal`'s yields a full matrix

### ISSUE THREAD

**Issue by nathanaelbosch** [MichielStock/Kronecker.jl#124]
With Kronecker.jl, this is the kind of operations that I want to do and the result that I'd expect:
```julia
using Kronecker

A = kronecker(rand(2, 2), rand(2, 2))
B = kronecker(I(2), rand(2, 2))
A*B isa Kronecker.KroneckerProduct
```
But if I multiply with a `KroneckerProduct` of `Diagonal` matrices, I suddently get a full matrix back:
```julia
C = kronecker(I(2), I(2))
A*C isa Matrix
```
This is unexpected and probably a bug.

**Comment by MichielStock** [comment]
Hi Nathan,

Thanks for noting this. It is addressed in #125. We are a bit too quick to turn Kroneckers of diagonals into dense versions!

M

**Comment by nathanaelbosch** [comment]
Thanks for the quick fix!
